hpx/parallel/datapar/execution_policy.hpp

```diff
@@ -289,44 +450,885 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
+          : std::true_type
+        {};
+        /// \endcond
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail
+    {
+        /// \cond NOINTERNAL
+        template <>
+        struct is_sequential_execution_policy<dataseq_execution_policy>
+          : std::true_type
+        {};
+
+        template <>
+        struct is_sequential_execution_policy<dataseq_task_execution_policy>
+          : std::true_type
+        {};
+
+        template <typename Executor, typename Parameters>
+        struct is_sequential_execution_policy<
+                dataseq_execution_policy_shim<Executor, Parameters> >
+          : std::true_type
+        {};
+
+        template <typename Executor, typename Parameters>
+        struct is_sequential_execution_policy<
+                dataseq_task_execution_policy_shim<Executor, Parameters> >
+          : std::true_type
+        {};
+        /// \endcond
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail
+    {
+        /// \cond NOINTERNAL
+        template <>
+        struct is_async_execution_policy<dataseq_task_execution_policy>
+          : std::true_type
+        {};
+
+        template <typename Executor, typename Parameters>
+        struct is_async_execution_policy<
+                dataseq_task_execution_policy_shim<Executor, Parameters> >
+          : std::true_type
+        {};
+
+        template <>
+        struct is_async_execution_policy<datapar_task_execution_policy>
+          : std::true_type
+        {};
+
+        template <typename Executor, typename Parameters>
+        struct is_async_execution_policy<
+                datapar_task_execution_policy_shim<Executor, Parameters> >
+          : std::true_type
+        {};
+        /// \endcond
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail
+    {
+        /// \cond NOINTERNAL
+        template <>
+        struct is_parallel_execution_policy<datapar_execution_policy>
+          : std::true_type
+        {};
+
+        template <>
+        struct is_parallel_execution_policy<datapar_task_execution_policy>
+          : std::true_type
+        {};
+
+        template <typename Executor, typename Parameters>
+        struct is_parallel_execution_policy<
+                datapar_execution_policy_shim<Executor, Parameters> >
+          : std::true_type
+        {};
+
+        template <typename Executor, typename Parameters>
+        struct is_parallel_execution_policy<
+                datapar_task_execution_policy_shim<Executor, Parameters> >
+          : std::true_type
+        {};
+        /// \endcond
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail
+    {
+        /// \cond NOINTERNAL
+        template <>
+        struct is_datapar_execution_policy<dataseq_execution_policy>
```

Comment:
`is_datapar_execution_policy` should probably be renamed to something else, I'm not sure what name to use, though. Any ideas?